base/loading.jl

```diff
@@ -1338,7 +1338,7 @@ function compilecache(pkg::PkgId, path::String)
             chmod(tmppath, filemode(path) & 0o777)
 
             # this is atomic according to POSIX:
-            rename(tmppath, cachefile)
+            rename(tmppath, cachefile; force=true)
```

Comment:
This could simply be changed to `mv(tmppath, cachefile; force = true)`, and then the changes to `rename()` are unnecessary.

But I'm not too happy with this solution because the reason this was added in the first place was for distributed execution on a shared filesystem, so that processes wouldn't error out because of racing to create the cache. With this fix, every process will overwrite the cache file -- the intended behavior was for all the processes that lost the race to silently fail to rename their temporary cache to the primary cache.

I was thinking of adding the following before the `finally`:
```julia
catch e
    if !isa(e, ArgumentError)
        throw(e)
    end
```

Or something like that but it doesn't matter much I suppose; we don't save on the system calls with the intended way and `rename` is atomic so it shouldn't break anything.

Any thoughts on this @KristofferC?

Reply:
@kpamnany  Note the current error in the linked issue does not involve distributed execution. The error crops up upon *any* recompilation of  packages, which makes it impossible to use `Pkg` on master without manually deleting the procompilation files.

Reply:
Yes, I understand. The swiftest fix is to change the `rename` to `mv` with `force = true` as I said.

Reply:
Got it, might be best if you could open a PR to discuss your proposed improvement over this stop-gap solution.

Reply:
On inspection, `mv()` uses `checkfor_mv_cp_cptree()` _before_ `rename()` and that function does an `rm()`, thus destroying the atomicity of `rename()` which is what we wanted in the first place. So your fix here is good.

Reply:
I'm going to merge this in the meantime because I can't currently develop and use Julia packages on master with this bug.

It's probably worthwhile to explore something like in  https://github.com/JuliaLang/julia/pull/36638#discussion_r453820564 in  a separate PR